addons/hw_drivers/controllers/driver.py

```diff
@@ -1,432 +1,326 @@
+            reboot = 'noreboot'
+            path = os.getenv("HOME") + '/odoo/addons/point_of_sale/tools/posbox/configuration/connect_to_server.sh'
+            subprocess_call([path, url, iotname, token, reboot])
+            m.send_alldevices()
+            f = open('/var/www/True.jpg','rb')
+            return f.read()
 
 #----------------------------------------------------------
-# Bluetooth
+# Drivers
 #----------------------------------------------------------
-class GattBtManager(gatt.DeviceManager):
-
-    def device_discovered(self, device):
-        # TODO: need some kind of updated_devices mechanism or not?
-        path = "bt_%s" % (device.mac_address,)
-        if path not in drivers:
-            for driverclass in btdrivers:
-                d = driverclass(device = device, manager=self)
-                if d.supported():
-                    drivers[path] = d
-                    d.connect()
-                    send_iot_box_device(False)
 
+drivers = []
+bt_devices = {}
+iot_devices = {}
 
-class BtManager(Thread):
-    gatt_manager = False
-
-    def run(self):
-        dm = GattBtManager(adapter_name='hci0')
-        self.gatt_manager = dm
-        dm.start_discovery()
-        dm.run()
-
-#----------------------------------------------------------
-# Bluetooth drivers
-#----------------------------------------------------------
-btdrivers = []
-
-class BtMetaClass(type):
+class MetaClass(type):
     def __new__(cls, clsname, bases, attrs):
-        newclass = super(BtMetaClass, cls).__new__(cls, clsname, bases, attrs)
-        btdrivers.append(newclass)
+        newclass = super(MetaClass, cls).__new__(cls, clsname, bases, attrs)
+        drivers.append(newclass)
         return newclass
 
-
-class BtDriver(Driver, metaclass=BtMetaClass):
-
+class Driver(Thread, metaclass=MetaClass):
+    connection_type = ""
 
     def __init__(self, device, manager):
-        super(BtDriver, self).__init__()
+        super(Driver, self).__init__()
         self.dev = device
         self.manager = manager
-        self.value = ''
+        self.value = ""
+        self.data = ""
         self.gatt_device = False
 
-    def disconnect(self):
-        path = "bt_%s" % (self.dev.mac_address,)
-        del drivers[path]
-
     def get_name(self):
-        return self.dev.alias()
+        return ''
+
+    def get_identifier(self):
+        return ''
+
+    def get_connection(self):
+        return ''
+
+    def get_message(self):
+        return ''
 
     def value(self):
         return self.value
 
+    def supported(self):
+        pass
+
     def action(self, action):
         pass
 
-    def get_connection(self):
-        return 'bluetooth'
+    def disconnect(self):
+        del iot_devices[self.get_identifier()]
 
-    def connect(self):
-        pass
 
+#----------------------------------------------------------
+# Device manager
+#----------------------------------------------------------
+
+class DeviceManager():
```

Comment:
sometimes your classes inherit from object, sometimes not
it doens't make a difference in python3 but you should be consistent
always inherit from object

Reply:
ping